bookwyrm/static/js/bookwyrm.js

```diff
@@ -368,4 +374,84 @@ let BookWyrm = new class {
             );
         }
     }
+
+    handleModalButton(element) {
+        const modalButton = element.currentTarget;
+        const targetModalId = modalButton.dataset.modalOpen;
+        const htmlElement = document.querySelector('html');
+        const modal = document.getElementById(targetModalId);
+        
+        if (!modal) {
+            return;
+        }
+
+        // Helper functions
+        function handleModalOpen(modalElement) {
+            htmlElement.classList.add('is-clipped');
+            modalElement.classList.add('is-active');
+            modalElement.getElementsByClassName('modal-card')[0].focus();
```

Comment:
Will this work for situations where there are multiple modals on a page?

Reply:
I think so, but I haven't tested. I tried to get `handleModalOpen` to work in the context of each specific modal. For this line, we focus the first element with that class that is inside the `modalElement` that has been passed, and that element is the one that's targeted by the value of `data-modal-open`